stf.core/src/stf.core/net/adoptopenjdk/stf/extensions/core/StfCoreExtension.java

```diff
@@ -1459,6 +1459,31 @@ public void verifyJavaArgsContains(Stage stage, String... mandatoryArgs) throws
 					+ "Actual Java arguments for this stage are \"" + convertToString(actualArgs) + "\"");
 		}
 	}
+	
+	
+	/**
+	 * Checks if that java processes using the primary JVM contains the given JVM option
+	 * 
+	 * @param stage is the stage whose arguments should be checked.
+	 * @param argsToCheck contains 1 java arguments.
+	 */
+	public boolean isJavaArgPresent(Stage stage, String argsToCheck) throws StfException {
+		if (stage == Stage.INITIALISATION) {
+			throw new StfException("Java argument verification is not valid for the initialisation stage.");
+		}
+
+		// Get hold of the mandatory and actual java arguments
+		List<String> argsToCheckList = Arrays.asList(argsToCheck);
+		ArrayList<String> actualArgs = getActiveJavaArgs(stage);
+
+		// Subtract the actual args from the mandatory. 
+		// The remaining args are mandatories which haven't been given a value.
+		Set<String> argsToCheckSet = new LinkedHashSet<String>(argsToCheckList);
+		Set<String> actualArgsSet = new LinkedHashSet<String>(actualArgs);
+		argsToCheckSet.removeAll(actualArgsSet);
```

Comment:
Given it a second thought, probably just check if actualArgsSet contains argsToCheck.

Reply:
Yep. Actually I've made that change. Tested here: /view/Test_grinder/job/Grinder_Advanced/1249/console

Thanks Sophia!